source/base/subscriptor.cc

```diff
@@ -175,7 +158,6 @@ Subscriptor::subscribe(const char *id) const
 #ifdef DEBUG
   if (object_info == 0)
     object_info = &typeid(*this);
-  Threads::Mutex::ScopedLock lock (subscription_lock);
   ++counter;
```

Comment:
This is correct: the next part is never run if we compile with threads (we never insert anything into the map as long as `DEAL_II_WITH_THREADS` is defined).

I should update the relevant comment in `subscriptor.h`:

```cpp
 /*
 [...]
 * @note Due to a problem with <tt>volatile</tt> declarations, this additional
 * feature is switched off if multithreading is used.
 [...]
 */
```

We should probably reword this since we don't use volatile values.